packages/logs/src/boot/preStartLogs.ts

```diff
@@ -43,6 +43,13 @@ export function createPreStartStrategy(
   const userContext = buildUserContextManager()
   bufferContextCalls(userContext, CustomerContextKey.userContext, bufferApiCalls)
 
+  // TODO next major: remove cacheUrlContext, less precision in pre-init URL context is acceptable, and RUM is already less precise
+  // `location` is undefined during server-side rendering (e.g., in Node)
+  const location = getGlobalObject().location
+  if (location) {
+    cacheUrlContext(location)
```

Comment:
Caches the url context for pre-start calls to prevent breaking changes

Reply:
> less precision in pre-init URL context is acceptable

so it is a breaking change, no? 
FMU, with this approach, we may associate logs with wrong URLs whereas it is not the case for now.

Instead of caching the init value, would it make sense to track a url context history?

Reply:
Good point! Done: https://github.com/DataDog/browser-sdk/pull/3653/commits/5d285025568463ba2b32275ba9ab5f0de74af0fb